Authentication issues

The new method of access broke the whole thing for me. Sync workflows have been failing since.
The errors are all related to authentication, caused by the captcha and 2-step authentication page that pops up on each attempt to sign in on a new device, and that prompts you to enter an OTP received by email.

![image](https://github.com/cecobask/imdb-trakt-sync/assets/73847973/2e866630-9ea1-495c-9044-cdde32e43244)

I think providing your IMDB username should be enough to access your public IMDB lists, needless of signing up and the issues that come up with it.

Thank you for such a quick reply. I fixed the privacy issue immediately and moved PW and email to secrets.  I'm now facing the 'other location' log in issue that I've seen decribed. I get the emails from IMDB with a single use login authentication code to use (but obviously I'm not 'present' in the attempts to us it. I have manually logged in and out of IMDB to try and clear the blockage. The workflow records this:

 ##[debug]/usr/bin/bash -e /home/runner/work/_temp/00c57a4d-971f-4084-a038-74a41eba74cf.sh
{"time":"2024-07-10T07:55:36.095019667Z","level":"INFO","source":{"function":"github.com/cecobask/imdb-trakt-sync/pkg/client.NewIMDbClient","file":"/home/runner/work/imdb-trakt-sync/imdb-trakt-sync/pkg/client/imdb.go","line":69},"msg":"launched new browser instance","url":"ws://127.0.0.1:46405/devtools/browser/0b617ca4-[29](https://github.com/Donostio/imdb-trakt-sync/actions/runs/9861235645/job/27255832857#step:6:30)7f-4c8c-88b2-bdf8e6b84558","headless":true,"trace":false}
Error: error creating syncer: failure initialising imdb client: failure authenticating user: failure authenticating as captcha prompt appeared
make: *** [Makefile:10: sync] Error 1
Error: Process completed with exit code 2

Set it all up again. It looks like I;m still getting stuck on IMDB having a captcha. Is there anything I can tweak to make it work? Thank you

_Run make sync
##[debug]/usr/bin/bash -e /home/runner/work/_temp/1d1f[28](https://github.com/Donostio/imdb-trakt-sync/actions/runs/9872845088/job/27264532536#step:6:29)1e-e126-4905-a4e6-0788964ae5e9.sh
{"time":"2024-07-10T11:08:36.264635407Z","level":"INFO","source":{"function":"github.com/cecobask/imdb-trakt-sync/pkg/client.NewIMDbClient","file":"/home/runner/work/imdb-trakt-sync/imdb-trakt-sync/pkg/client/imdb.go","line":68},"msg":"launched new browser instance","url":"ws://127.0.0.1:41917/devtools/browser/8c06f0bc-afed-48a7-9a7c-dc80eca5adc7","headless":true,"trace":false}
Error: error creating syncer: failure initialising imdb client: failure authenticating user: failure authenticating as captcha prompt appeared
make: *** [Makefile:10: sync] Error 1
Error: Process completed with exit code 2.
##[debug]Finishing: Sync_

Run make sync
##[debug]/usr/bin/bash -e /home/runner/work/_temp/b60b6e4b-4276-4f9a-ada2-d19fee[28](https://github.com/Donostio/imdb-trakt-sync/actions/runs/9873811471/job/27267203470#step:6:29)41a2.sh
{"time":"2024-07-10T12:11:20.268075141Z","level":"INFO","source":{"function":"github.com/cecobask/imdb-trakt-sync/pkg/client.NewIMDbClient","file":"/home/runner/work/imdb-trakt-sync/imdb-trakt-sync/pkg/client/imdb.go","line":68},"msg":"launched new browser instance","url":"ws://127.0.0.1:4[29](https://github.com/Donostio/imdb-trakt-sync/actions/runs/9873811471/job/27267203470#step:6:30)91/devtools/browser/1a5e58d1-f175-4c5d-b661-cfe48815f320","headless":true,"trace":false}
Error: error creating syncer: failure initialising imdb client: failure authenticating user: failure doing selector race: context deadline exceeded
make: *** [Makefile:10: sync] Error 1
Error: Process completed with exit code 2.
##[debug]Finishing: Sync

--- a/pkg/client/imdb.go
+++ b/pkg/client/imdb.go
@@ -481,9 +481,12 @@ func (c *IMDbClient) lidsScrape() ([]string, error) {
 	if tab, err = navigateAndValidateResponse(tab, imdbPathBase+imdbPathLists); err != nil {
 		return nil, fmt.Errorf("failure navigating and validating response: %w", err)
 	}
-	listCountDiv, err := tab.Element("div[data-testid='list-page-mc-total-items']")
+	hasLists, listCountDiv, err := tab.Has("div[data-testid='list-page-mc-total-items']")
 	if err != nil {
 		return nil, fmt.Errorf("failure finding list count div: %w", err)
+	}
+	if !hasLists {
+		return make([]string, 0), nil
 	}
 	listCountText, err := listCountDiv.Text()
 	if err != nil {
@@ -630,7 +633,7 @@ func navigateAndValidateResponse(tab *rod.Page, url string) (*rod.Page, error) {
 	if status := event.Response.Status; status != http.StatusOK {
 		return nil, fmt.Errorf("navigating to %s produced %d status", url, status)
 	}
-	if err := tab.WaitStable(time.Second); err != nil {
+	if err := tab.WaitLoad(); err != nil {
 		return nil, fmt.Errorf("failure waiting for tab to load: %w", err)
 	}
 	return tab, nil

